lib/cloud/awsconfig/awsconfig.go

```diff
@@ -80,6 +81,9 @@ type AssumeRole struct {
 	// Tags is a list of STS session tags to pass when assuming the role.
 	// https://docs.aws.amazon.com/IAM/latest/UserGuide/id_session-tags.html
 	Tags map[string]string `json:"tags,omitempty"`
+	// Duration is the expirity duration of the generated credentials. Empty
+	// value will use the AWS SDK default expiration time.
+	Duration time.Duration `json:"duraiton,omitempty"`
```

Comment:
Typo on the JSON tag here. 

Also note that omitempty probably doesn't work like you expect for duration values. (This is why Go 1.24 has omitzero). You probably need to make this a pointer for now.

Reply:
Oof, you're right. Great catch Gavin.

So the `duraiton` typo is the only action needed here.